nilearn/datasets/_utils.py

```diff
@@ -610,7 +610,7 @@ def fetch_single_file(
                     ):
                         raise OSError("Server does not support resuming")
                     initial_size = local_file_size
-                    with open(local_file, "ab") as fh:
+                    with local_file.open("ab") as fh:
```

Comment:
I think this is a bug. These lines do not seem to be covered by the current tests, and `local_file` is set to `None` at Line ~574 and never changed after, so as far as I can tell the file opening should fail regardless of which function/method is used. I guess it is not that easy to test because it requires a partially downloaded file.

I am not familiar with what the function is supposed to do but if someone confirms that this is a bug then I can create an issue for it.

Reply:
yeah let's open a separate issue: did a bit git-archeology (gitcheology?)

it seems that the relevant changes happened a long time ago

https://github.com/nilearn/nilearn/blame/893b0fcc9b819254efc1ecad659a075b33fd2867/nilearn/datasets/utils.py#L576